BlockEntity/BETrough.cs

```diff
@@ -392,41 +392,40 @@ public override void GetBlockInfo(IPlayer forPlayer, StringBuilder dsc)
             dsc.AppendLine(Lang.Get("Portions: {0}", fillLevel));
 
             ItemStack contentsStack = config.Content.ResolvedItemstack ?? ResolveWildcardContent(config, forPlayer.Entity.World);
-            
-            if (contentsStack != null)
-            {
-                var cobj = contentsStack.Collectible;
-                var foodCat = cobj.NutritionProps?.FoodCategory ?? EnumFoodCategory.NoNutrition;
-                var foodTags = cobj.Attributes?["foodTags"].AsArray<string>();
 
-                dsc.AppendLine(Lang.Get(contentsStack.GetName()));
+            if (contentsStack == null) return;
+
+            var cobj = contentsStack.Collectible;
+            var foodCat = cobj.NutritionProps?.FoodCategory ?? EnumFoodCategory.NoNutrition;
+            var foodTags = cobj.Attributes?["foodTags"].AsArray<string>();
 
-                HashSet<string> creatureNames = new HashSet<string>();
-                foreach (var entityType in Api.World.EntityTypes)
+            dsc.AppendLine(Lang.Get(contentsStack.GetName()));
+
+            HashSet<string> creatureNames = new HashSet<string>();
+            foreach (var entityType in Api.World.EntityTypes)
+            {
+                var attr = entityType.Attributes;
+                if (attr == null || attr["creatureDiet"].Exists == false) continue;

```

Comment:
Just FYI this could simply have been written as `if (attr?["creatureDiet"].Exists != true) continue;` instead to not need the null checking separated. :)

Reply:
Thanks, new language + framework means I don't have the most concise terms yet!

Reply:
Oh no worries I wasn’t really asking you to change it anyway.  I always have to clean the code up from PRs anyway when I merge them in I was more just making you aware for your own sake.  Can you place squash the commits?  It’s a pain to work with PRs that have more than 1 commit. :)